tools/base/libboard/board_vc/board_vc.c

```diff
@@ -0,0 +1,746 @@
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get token object");
+		resval = res;
+		goto out_destroy_bsid;
+	}
+
+	res = fpgaTokenGetObject(token, SYSFS_PKVL_STATUS, &status_object, FPGA_OBJECT_GLOB);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get token object");
+		resval = res;
+		goto out_destroy_poll;
+	}
+
+	res = fpgaObjectRead64(bsid_object, &bs_id, 0);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to read object ");
+		resval = res;
+		goto out_destroy_status;
+	}
+
+	*fpga_mode = (bs_id >> 32) & 0xf;
+
+	res = fpgaObjectRead64(poll_mode_object, &poll_mode, 0);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to read object ");
+		resval = res;
+		goto out_destroy_status;
+	}
+
+	res = fpgaObjectRead64(status_object, &status, 0);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to read object ");
+		resval = res;
+		goto out_destroy_status;
+	}
+
+	pkvl_info->polling_mode = (uint32_t)poll_mode;
+	pkvl_info->status = (uint32_t)status;
+
+out_destroy_status:
+	res = fpgaDestroyObject(&status_object);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to destroy object");
+	}
+
+out_destroy_poll:
+	res = fpgaDestroyObject(&poll_mode_object);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to destroy object");
+	}
+
+out_destroy_bsid:
+	res = fpgaDestroyObject(&bsid_object);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to destroy object");
+	}
+
+	return resval;
+}
+
+// Read PHY group information
+fpga_result read_phy_group_info(fpga_token token,
+			fpga_phy_group_info *group_info,
+			int *group_num)
+{
+	fpga_result res                 = FPGA_OK;
+	fpga_result resval              = FPGA_OK;
+	char path[SYSFS_MAX_SIZE]       = { 0 };
+	char cdevid[CDEV_ID_SIZE]       = { 0 };
+	size_t i                        = 0;
+	int fd                          = 0;
+	uint32_t group_dev_count        = 0;
+	uint32_t obj_size               = 0;
+	fpga_object dev_obj[SYSFS_MAX_SIZE];
+	fpga_phy_group_info info;
+	fpga_object group_object;
+	fpga_object group_dev_object;
+
+
+	if (group_num == NULL ) {
+		FPGA_ERR("Invalid Input parameters");
+		return FPGA_INVALID_PARAM;
+	}
+
+	res = fpgaTokenGetObject(token, SYSFS_PHY_GROUP_INFO,
+		&group_object, FPGA_OBJECT_GLOB);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get token object");
+		return FPGA_NOT_FOUND;
+	}
+
+	res = fpgaTokenGetObject(token, SYSFS_PHY_GROUP_INFO_DEV,
+		&group_dev_object, FPGA_OBJECT_GLOB);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get token object");
+		resval = res;
+		goto out_destroy_group;
+	}
+
+	res = fpgaObjectGetSize(group_dev_object, &group_dev_count, FPGA_OBJECT_GLOB);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get token object size");
+		resval = res;
+		goto out_destroy_group_dev;
+	}
+
+	// Return number of group.
+	if (group_info == NULL &&
+		group_num != NULL) {
+
+		*group_num = group_dev_count;
+		resval = FPGA_OK;
+		goto out_destroy_group_dev;
+	}
+
+
+	for (i = 0; i < group_dev_count; i++) {
```

Comment:
what if group_dev_count is equal to or greater than the number of elements in the array dev_obj?

Reply:
yes 
number of elements in  array of dev_obj after enum with object api

Reply:
if group_dev_count greater then input fpga_phy_group_info size return error

fixed